cpp/src/knn/knn_opg_common.cuh

```diff
@@ -525,7 +525,7 @@ void copy_label_outputs_from_index_parts(opg_knn_param<in_t, ind_t, dist_t, out_
   std::vector<out_t*> parts_h(n_parts);
   device_buffer<out_t*> parts_d(handle.get_device_allocator(), handle.get_stream(), n_parts);
   for (int o = 0; o < params.n_outputs; o++) {
-    for (int p = 0; p < n_parts; p++) {
+    for (uint64_t p = 0; p < n_parts; p++) {
```

Comment:
I think both this an `n_parts` should probably just be `std::size_t`. You can update the `n_parts` declaration at line 521.